### ISSUE TITLE
Update type signature of `RpcProviderModule` to implement Web3's `AbstractProvider`

### ISSUE THREAD

**Issue by smithki** [magiclabs/magic-js#83]
### ✅ Prerequisites

- [x] Did you perform a cursory search of open issues? Is this feature already requested elsewhere?
- [x] Are you reporting to the correct repository (`magic-sdk`)?

### ✨ Feature Request

Currently, to use Web3 features in TypeScript, the `magic.rpcProvider` member must be cast to `any`, like this:

```ts
new Web3(magic.rpcProvider as any);
```

But, now that https://github.com/ethereum/web3.js/issues/3380 is resolved, we can implement the Web3 `AbstractProvider` interface to gain strong typing. Once this is implemented, consumers of `rpcProvider` can simply do this:

```ts
new Web3(magic.rpcProvider); // No compiler error! Yay! 🚀
```

**Comment by sebastiandine** [comment]
This issue still exists in the latest version. The workaround is still applicable but it would be nice to see this completely fixed in future releases.

**Comment by smithki** [comment]
We are open to a community PR @sebastiandine!

**Comment by smithki** [comment]
:rocket: Issue was released in [`@magic-sdk/provider@5.1.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/provider%405.1.0), [`@magic-sdk/react-native@5.1.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/react-native%405.1.0), [`magic-sdk@5.1.0`](https://github.com/magiclabs/magic-js/releases/tag/magic-sdk%405.1.0) :rocket:

**Comment by smithki** [comment]
:rocket: Issue was released in [`@magic-sdk/provider@5.1.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/provider%405.1.0), [`@magic-sdk/react-native@5.1.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/react-native%405.1.0), [`magic-sdk@5.1.0`](https://github.com/magiclabs/magic-js/releases/tag/magic-sdk%405.1.0) :rocket:

**Comment by smithki** [comment]
:rocket: Issue was released in [`@magic-sdk/commons@2.0.0-next.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/commons%402.0.0-next.0), [`@magic-sdk/provider@6.0.0-next.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/provider%406.0.0-next.0), [`@magic-sdk/react-native@6.0.0-next.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/react-native%406.0.0-next.0), [`@magic-sdk/types@5.0.0-next.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/types%405.0.0-next.0), [`magic-sdk@6.0.0-next.0`](https://github.com/magiclabs/magic-js/releases/tag/magic-sdk%406.0.0-next.0) :rocket:

**Comment by smithki** [comment]
:rocket: Issue was released in [`@magic-sdk/commons@3.0.0-dgerrells-ch42053-sms-feature.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/commons%403.0.0-dgerrells-ch42053-sms-feature.0), [`@magic-sdk/provider@7.0.0-dgerrells-ch42053-sms-feature.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/provider%407.0.0-dgerrells-ch42053-sms-feature.0), [`@magic-sdk/react-native@7.0.0-dgerrells-ch42053-sms-feature.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/react-native%407.0.0-dgerrells-ch42053-sms-feature.0), [`@magic-sdk/types@6.0.0-dgerrells-ch42053-sms-feature.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/types%406.0.0-dgerrells-ch42053-sms-feature.0), [`magic-sdk@7.0.0-dgerrells-ch42053-sms-feature.0`](https://github.com/magiclabs/magic-js/releases/tag/magic-sdk%407.0.0-dgerrells-ch42053-sms-feature.0) :rocket:

**Comment by smithki** [comment]
:rocket: Issue was released in [`@magic-sdk/commons@3.0.0-dgerrells-ch42053-sms-feature.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/commons%403.0.0-dgerrells-ch42053-sms-feature.0), [`@magic-sdk/provider@7.0.0-dgerrells-ch42053-sms-feature.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/provider%407.0.0-dgerrells-ch42053-sms-feature.0), [`@magic-sdk/react-native@7.0.0-dgerrells-ch42053-sms-feature.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/react-native%407.0.0-dgerrells-ch42053-sms-feature.0), [`@magic-sdk/types@6.0.0-dgerrells-ch42053-sms-feature.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/types%406.0.0-dgerrells-ch42053-sms-feature.0), [`magic-sdk@7.0.0-dgerrells-ch42053-sms-feature.0`](https://github.com/magiclabs/magic-js/releases/tag/magic-sdk%407.0.0-dgerrells-ch42053-sms-feature.0) :rocket:

**Comment by smithki** [comment]
:rocket: Issue was released in [`@magic-sdk/commons@3.0.0-dgerrells-ch42053-sms-feature.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/commons%403.0.0-dgerrells-ch42053-sms-feature.0), [`@magic-sdk/provider@7.0.0-dgerrells-ch42053-sms-feature.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/provider%407.0.0-dgerrells-ch42053-sms-feature.0), [`@magic-sdk/react-native@7.0.0-dgerrells-ch42053-sms-feature.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/react-native%407.0.0-dgerrells-ch42053-sms-feature.0), [`@magic-sdk/types@6.0.0-dgerrells-ch42053-sms-feature.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/types%406.0.0-dgerrells-ch42053-sms-feature.0), [`magic-sdk@7.0.0-dgerrells-ch42053-sms-feature.0`](https://github.com/magiclabs/magic-js/releases/tag/magic-sdk%407.0.0-dgerrells-ch42053-sms-feature.0) :rocket:
